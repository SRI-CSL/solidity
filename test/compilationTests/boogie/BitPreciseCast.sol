pragma solidity ^0.4.23;

contract BitPreciseCast {

    function explicitCasts() public pure {
        uint8 x = 255;
        assert(int8(x) == -1);
        uint16 y = 257;
        assert(uint8(y) == 1);
        int8 z = -3;
        assert(uint8(z) == 253);
    }
}