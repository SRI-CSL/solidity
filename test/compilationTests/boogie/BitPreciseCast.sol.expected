Verifying BitPreciseCast::__verifier_main: OK
Verifying BitPreciseCast::[implicit_constructor]: OK
No errors found.
